Assets/MRTK/Core/Utilities/StandardShader/ClippingPrimitive.cs

```diff
@@ -127,57 +136,51 @@ public bool IsDirty
             set => isDirty = value;
         }
 
+        /// <summary>
+        /// Public access for the renderer(s) that should be affected by the primitive.
+        /// </summary>
+        public List<Renderer> Renderers => renderersCache;
+
         /// <summary>
         /// Adds a renderer to the list of objects this clipping primitive clips.
         /// </summary>
-        /// <param name="_renderer">The renderer to add.</param>
-        public void AddRenderer(Renderer _renderer)
+        /// <param name="renderer">The renderer to add.</param>
+        public void AddRenderer(Renderer renderer)
         {
-            if (_renderer != null)
+            if (renderer != null)
             {
-                if (!renderers.Contains(_renderer))
+                if (!renderers.ContainsKey(renderer))
                 {
-                    renderers.Add(_renderer);
-                }
+                    var materialInstance = renderer.EnsureComponent<MaterialInstance>();
+                    if (!materialInstance)
```

Comment:
I don't think EnsureComponent will return null? Unless AddComponent fails.

Reply:
It's just a sanity check. As you said - It can return null if AddComponent fails. It may fail if gameObject has already been destroyed.